nsv13/code/controllers/subsystem/overmap_mode.dm

```diff
@@ -0,0 +1,373 @@
+
+/datum/controller/subsystem/overmap_mode/proc/difficulty_calc()
+	var/players = get_active_player_count(TRUE, FALSE, FALSE) //Check how many players are still alive
+	switch(players)
+		if(0 to 10)
+			mode.difficulty = 1
+		if(10 to 20)
+			mode.difficulty = 2
+		if(20 to 30)
+			mode.difficulty = 3
+		if(30 to 40)
+			mode.difficulty = 4
+		if(40 to INFINITY)
+			mode.difficulty = 5
+
+	mode.difficulty += escalation //Our admin adjustment
+	if(mode.difficulty <= 0)
+		mode.difficulty = 1
+
+/datum/overmap_gamemode
+	var/name = null											//Name of the gamemode type
+	var/desc = null											//Description of the gamemode for ADMINS
+	var/brief = null										//Description of the gamemode for PLAYERS
+	var/config_tag = null									//Do we have a tag?
+	var/selection_weight = 0								//Used to determine the chance of this gamemode being selected
+	var/required_players = 0								//Required number of players for this gamemode to be randomly selected
+	var/difficulty = null									//Difficulty of the gamemode as determined by player count / abus abuse: 1 is minimum, 10 is maximum
+	var/starting_system = null								//Here we define where our player ships will start
+	var/starting_faction = null 							//Here we define which faction our player ships belong
+	var/objective_reminder_setting = REMINDER_OBJECTIVES	//0 - Objectives reset remind. 1 - Combat resets reminder. 2 - Combat delays reminder. 3 - Disables reminder
+	var/objective_reminder_interval = 15 MINUTES			//Interval between objective reminders
+	var/combat_delay = 0									//How much time is added to the reminder timer
+	var/list/objectives = list()							//The actual gamemode objectives go here
+	var/whitelist_only = FALSE								//Can only be selected through map bound whitelists
+
+	//Reminder messages
+	var/reminder_origin = "Naval Command"
+	var/reminder_one = "Case 1"
+	var/reminder_two = "Case 2"
+	var/reminder_three = "Case 3"
+	var/reminder_four = "Case 4"
+	var/reminder_five = "Case 5"
+
+/datum/overmap_gamemode/proc/consequence_one()
+
+
+/datum/overmap_gamemode/proc/consequence_two()
+
+
+/datum/overmap_gamemode/proc/consequence_three()
+
+
+/datum/overmap_gamemode/proc/consequence_four()
+	var/datum/faction/F = SSstar_system.faction_by_id(starting_faction)
+	F.lose_influence(100)
+
+/datum/overmap_gamemode/proc/consequence_five()
+	//Hotdrop O'Clock
+	var/datum/star_system/target = SSstar_system.find_main_overmap().current_system
+	priority_announce("Faction Hunter something something Hotdrop") //need a faction message
+	var/datum/fleet/F = new /datum/fleet/something() //need a fleet
+	target.fleets += F
+	F.current_system = target
+	F.assemble(target)
+
+
+
+/datum/overmap_gamemode/proc/check_completion() //This gets called by checking the communication console/modcomp program + automatically once every 10 minutes
+	//First we try to check completion on each objective
+	for(var/datum/overmap_objective/O in objectives)
+		O.check_completion()
```

Comment:
Since bonus objectives get added to the mode, can just check all objectives at once